dll/cpl/mmsys/audio.c

```diff
@@ -146,6 +161,54 @@ InitAudioDlg(HWND hwnd)
     }
 }
 
+VOID
+UpdateRegistryString(HWND hwnd, INT ctrl, LPWSTR key, LPWSTR value)
+{
+    HWND hwndCombo = GetDlgItem(hwnd, ctrl);
+    INT CurSel = SendMessage(hwndCombo, CB_GETCURSEL, 0, 0);
+    WCHAR SelectedDevice[MAX_PATH] = {0};
+    HKEY hKey;
+
+    if (CurSel == CB_ERR)
+        return;
+
+    SendMessageW(hwndCombo, CB_GETLBTEXT, CurSel, (LPARAM)&SelectedDevice);
```

Comment:
Hopefully there won't be any buffer overrun if the string is too long? xDD

Reply:
It already uses the string with length of `MAX_PATH` when populating the combo box list on initialization, so should not happen :)